src/program.cpp

```diff
@@ -741,7 +390,7 @@ void program::perf_report(std::ostream& os, std::size_t n, parameter_map params)
 void program::debug_print() const { std::cout << *this << std::endl; }
 void program::debug_print(instruction_ref ins) const
 {
-    if(ins == this->end())
+    if(ins == this->impl->modules["main"].end())
```

Comment:
I would assume this function(since it in the program) would search all modules for the instruction.

Reply:
fixed